main.nf

```diff
@@ -311,17 +338,19 @@ process miRBasePostAlignment {
     maxRetries 3
     maxErrors '-1'
 
-    publishDir "${params.outdir}/miRBaseCounts", mode: 'copy'
+    publishDir "${params.out}", mode: 'copy', saveAs: this.&rule_step5
```

Comment:
Ok, so this is passed to the function? In that case, can we pass `$params.outdir` and then append to that in the function call instead.. I don't see why we need `$params.out` as well? In fact, that variable doesn't seem to be used by anything else in the script, so can we remove it entirely?

That wasn't very clear. Proposed code:
```groovy
publishDir $params.outdir, mode: 'copy', saveAs: this.&rule_step5
```
```groovy
def rule_step5(file){
      if ( file.contains("mature") )
        return "/bowtie_miRBase_mature/$file"

      if ( file.contains("hairpin") )
        return "/bowtie_miRBase_hairpin/$file"
}
```

Better still, we could have a single function for both this process _and_ the edgeR process:

```groovy
publishDir "${params.outdir}/bowtie_miRBase", mode: 'copy', saveAs: this.&mature_hairpin_dirs
```
```groovy
publishDir "${params.outdir}/edgeR", mode: 'copy', saveAs: this.&mature_hairpin_dirs
```
```groovy
def mature_hairpin_dirs(file){
      if ( file.contains("mature") )
        return "/mature/$file"

      if ( file.contains("hairpin") )
        return "/hairpin/$file"
}
```